.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
+  - repo: https://github.com/ambv/black
+    rev: 22.3.0
+    hooks:
+    - id: black
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.2.3
+    hooks:
+    - id: nbqa-black
+    - id: nbqa-isort
+    - id: nbqa-flake8
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.12.0
+    hooks:
+    - id: isort
+  - repo: https://github.com/pycqa/flake8
```

Comment:
Lets switch to [ruff](https://github.com/astral-sh/ruff) instead of flake8, it's much faster and meshes fine with Diffrax.

Pretty simple to switch to, repo is https://github.com/charliermarsh/ruff-pre-commit and it needs `args: [--fix]`